### PR TITLE
Jackett: Update dotnet build for DSM 7.2

### LINKED ISSUE #5688: jackett can not start with dsm 7.2 beta (DSM 7.2-64216)
### Is this a new Bug?

- [X] I checkd that the bug hasn't been reported before

### Package Name

jackett

### Package Version

 0.20.1171-15

### Device Model

DS220+

### Device Architecture

x86_64

### Firmware Version

7.2-64216

### What happened?

```
Sun Apr  9 10:35:02 CST 2023
Starting jackett command env HOME=/volume1/@appdata/jackett PATH=/volume1/@appstore/jackett/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/syno/sbin:/usr/syno/bin:/usr/local/sbin:/usr/local/bin LD_LIBRARY_PATH=/volume1/@appstore/jackett/lib /volume1/@appstore/jackett/share/jackett --PIDFile /volume1/@appdata/jackett/jackett.pid
Process terminated. Couldn't find a valid ICU package installed on the system. Please install libicu using your package manager and try again. Alternatively you can set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support. Please see https://aka.ms/dotnet-missing-libicu for more information.
   at System.Environment.FailFast(System.String)
   at System.Globalization.GlobalizationMode+Settings..cctor()
   at System.Globalization.CultureData.CreateCultureWithInvariantData()
   at System.Globalization.CultureData.get_Invariant()
   at System.Globalization.CultureInfo..cctor()
   at System.Globalization.CultureInfo.get_InvariantCulture()
   at CommandLine.ParserSettings..ctor()
   at CommandLine.Parser..ctor(System.Action`1<CommandLine.ParserSettings>)
   at Jackett.Server.Program.Main(System.String[])
jackett is not running
```

### Reproduction steps

```bash
1. install 
2. start
```

### Install Log

```log
2023/04/09 10:34:58	install jackett 0.20.1171-15 Begin preinst
2023/04/09 10:34:58	Begin reload_inst_variables
2023/04/09 10:34:58	End reload_inst_variables
2023/04/09 10:34:58	Begin initialize_variables
2023/04/09 10:34:58	End initialize_variables
2023/04/09 10:34:58	===> Step preinst. USER=sc-jackett GROUP= SHARE_PATH=
2023/04/09 10:34:58	install jackett 0.20.1171-15 End preinst ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/mkdir -p /volume1/@appstore/jackett -m 755
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/mkdir -p /volume1/@appstore/jackett -m 755 ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/rm -rf /volume1/@appstore/jackett
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/rm -rf /volume1/@appstore/jackett ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/package /volume1/@appstore/jackett
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/package /volume1/@appstore/jackett ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/rm -rf /var/packages/jackett
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/rm -rf /var/packages/jackett ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/mkdir -p /var/packages/jackett -m 755
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/mkdir -p /var/packages/jackett -m 755 ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/touch /var/packages/jackett/installing
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/touch /var/packages/jackett/installing ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/INFO /var/packages/jackett/INFO
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/INFO /var/packages/jackett/INFO ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/rm -rf /var/packages/jackett/scripts
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/rm -rf /var/packages/jackett/scripts ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/scripts /var/packages/jackett/scripts
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/scripts /var/packages/jackett/scripts ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/rm -rf /var/packages/jackett/WIZARD_UIFILES
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/rm -rf /var/packages/jackett/WIZARD_UIFILES ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/WIZARD_UIFILES /var/packages/jackett/WIZARD_UIFILES
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/WIZARD_UIFILES /var/packages/jackett/WIZARD_UIFILES ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/rm -rf /var/packages/jackett/conf
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/rm -rf /var/packages/jackett/conf ret=[0]
2023/04/09 10:34:59	install jackett 0.20.1171-15 Begin /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/conf /var/packages/jackett/conf
2023/04/09 10:34:59	install jackett 0.20.1171-15 End /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/conf /var/packages/jackett/conf ret=[0]
2023/04/09 10:35:01	install jackett 0.20.1171-15 Begin postinst
2023/04/09 10:35:01	Begin reload_inst_variables
2023/04/09 10:35:01	End reload_inst_variables
2023/04/09 10:35:01	Begin initialize_variables
2023/04/09 10:35:01	End initialize_variables
2023/04/09 10:35:01	===> Step postinst. USER=sc-jackett GROUP= SHARE_PATH=
2023/04/09 10:35:01	Begin save_wizard_variables
2023/04/09 10:35:01	End save_wizard_variables
2023/04/09 10:35:01	Begin syno_sync_var_folder
2023/04/09 10:35:01	Install files from var folder
2023/04/09 10:35:01	/bin/rsync -avh --ignore-existing --remove-source-files /volume1/@appstore/jackett/var/ /volume1/@appdata/jackett
2023/04/09 10:35:01	sending incremental file list
2023/04/09 10:35:01	./
2023/04/09 10:35:01	.config/
2023/04/09 10:35:01	.config/Jackett/
2023/04/09 10:35:01	sent 119 bytes  received 27 bytes  292.00 bytes/sec
2023/04/09 10:35:01	total size is 0  speedup is 0.00
2023/04/09 10:35:01	/bin/rsync -avh --remove-source-files /volume1/@appstore/jackett/var/ /volume1/@appdata/jackett
2023/04/09 10:35:01	sending incremental file list
2023/04/09 10:35:01	sent 106 bytes  received 14 bytes  240.00 bytes/sec
2023/04/09 10:35:01	total size is 0  speedup is 0.00
2023/04/09 10:35:01	End syno_sync_var_folder
2023/04/09 10:35:01	install jackett 0.20.1171-15 End postinst ret=[0]
2023/04/09 10:35:02	install jackett 0.20.1171-15 Begin start-stop-status start
/var/packages/jackett/scripts/start-stop-status: line 115: 27597 Aborted                 (core dumped) ${SERVICE_COMMAND} >> ${OUT} 2>&1
2023/04/09 10:35:22	install jackett 0.20.1171-15 End start-stop-status start ret=[1]
2023/04/09 10:35:22	(system) trigger jackett 0.20.1171-15 Begin start-stop-status stop
2023/04/09 10:35:22	(system) trigger jackett 0.20.1171-15 End start-stop-status stop ret=[0]
```

### Service Log
```
Sun Apr  9 10:35:02 CST 2023
Starting jackett command env HOME=/volume1/@appdata/jackett PATH=/volume1/@appstore/jackett/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/syno/sbin:/usr/syno/bin:/usr/local/sbin:/usr/local/bin LD_LIBRARY_PATH=/volume1/@appstore/jackett/lib /volume1/@appstore/jackett/share/jackett --PIDFile /volume1/@appdata/jackett/jackett.pid
Process terminated. Couldn't find a valid ICU package installed on the system. Please install libicu using your package manager and try again. Alternatively you can set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support. Please see https://aka.ms/dotnet-missing-libicu for more information.
   at System.Environment.FailFast(System.String)
   at System.Globalization.GlobalizationMode+Settings..cctor()
   at System.Globalization.CultureData.CreateCultureWithInvariantData()
   at System.Globalization.CultureData.get_Invariant()
   at System.Globalization.CultureInfo..cctor()
   at System.Globalization.CultureInfo.get_InvariantCulture()
   at CommandLine.ParserSettings..ctor()
   at CommandLine.Parser..ctor(System.Action`1<CommandLine.ParserSettings>)
   at Jackett.Server.Program.Main(System.String[])
jackett is not running
```

_No response_

### Other Logs
```
2023-04-09T10:35:02+08:00 user coredump[27608]: Process jackett[27597](/volume1/@appstore/jackett/share/jackett) dumped core on signal [6]. Core file [/volume1/@jackett.synology_geminilake_220+.64216.core.gz]. Cmdline [/volume1/@appstore/jackett/share/jackett --PIDFile /volume1/@appdata/jackett/jackett.pid ]
2023-04-09T10:35:02+08:00 user coredump[27608]: Core file [/volume1/@jackett.synology_geminilake_220+.64216.core.gz] size [2782707]
```

```
2023/04/09 10:34:58 install jackett 0.20.1171-15 Begin preinst
2023/04/09 10:34:58 install jackett 0.20.1171-15 End preinst ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/mkdir -p /volume1/@appstore/jackett -m 755
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/mkdir -p /volume1/@appstore/jackett -m 755 ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/rm -rf /volume1/@appstore/jackett
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/rm -rf /volume1/@appstore/jackett ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/package /volume1/@appstore/jackett
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/package /volume1/@appstore/jackett ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/rm -rf /var/packages/jackett
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/rm -rf /var/packages/jackett ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/mkdir -p /var/packages/jackett -m 755
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/mkdir -p /var/packages/jackett -m 755 ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/touch /var/packages/jackett/installing
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/touch /var/packages/jackett/installing ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/INFO /var/packages/jackett/INFO
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/INFO /var/packages/jackett/INFO ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/rm -rf /var/packages/jackett/scripts
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/rm -rf /var/packages/jackett/scripts ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/scripts /var/packages/jackett/scripts
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/scripts /var/packages/jackett/scripts ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/rm -rf /var/packages/jackett/WIZARD_UIFILES
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/rm -rf /var/packages/jackett/WIZARD_UIFILES ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/WIZARD_UIFILES /var/packages/jackett/WIZARD_UIFILES
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/WIZARD_UIFILES /var/packages/jackett/WIZARD_UIFILES ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/rm -rf /var/packages/jackett/conf
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/rm -rf /var/packages/jackett/conf ret=[0]
2023/04/09 10:34:59 install jackett 0.20.1171-15 Begin /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/conf /var/packages/jackett/conf
2023/04/09 10:34:59 install jackett 0.20.1171-15 End /bin/mv -f /volume1/@tmp/synopkg/install.pLWtjP/conf /var/packages/jackett/conf ret=[0]
2023/04/09 10:35:00 Acquire systemd-unit for jackett when 0x0001 (ready)
2023/04/09 10:35:00 Acquire systemd-unit for jackett when 0x0001 (done)
2023/04/09 10:35:00 Acquire port-config for jackett when 0x0001 (ready)
2023/04/09 10:35:00 Acquire port-config for jackett when 0x0001 (done)
2023/04/09 10:35:01 install jackett 0.20.1171-15 Begin postinst
2023/04/09 10:35:01 install jackett 0.20.1171-15 End postinst ret=[0]
2023/04/09 10:35:01 install jackett: begin to start version 0.20.1171-15
2023/04/09 10:35:01 Acquire port-config for jackett when 0x0001 (ready)
2023/04/09 10:35:01 Acquire port-config for jackett when 0x0001 (done)
2023/04/09 10:35:02 install jackett 0.20.1171-15 Begin start-stop-status start
2023/04/09 10:35:22 install jackett 0.20.1171-15 End start-stop-status start ret=[1]
2023/04/09 10:35:22 Release port-config for jackett when 0x0001 (ready)
2023/04/09 10:35:22 Release port-config for jackett when 0x0001 (done)
2023/04/09 10:35:22 install jackett: start version 0.20.1171-15 failed, result 0
2023/04/09 10:35:22 install jackett 0.20.1171-15 successfully
2023/04/09 10:35:22 (system) trigger jackett 0.20.1171-15 Begin start-stop-status stop
2023/04/09 10:35:22 (system) trigger jackett 0.20.1171-15 End start-stop-status stop ret=[0]
2023/04/09 10:35:22 Release port-config for jackett when 0x0000 (ready)
2023/04/09 10:35:22 Release port-config for jackett when 0x0000 (done)
2023/04/09 10:35:22 enabled flag does not exist, skip apparmor unloading
```

_No response_

### PR DESCRIPTION
**Pull request by mreid-tt**
## Description

While running under DSM 7.2, Jackett throws an error `Couldn't find a valid ICU package installed on the system. Please install libicu using your package manager and try again`. This PR includes:

1. Fix dotnet dependencies for DSM 7.2
2. Exclusion of `alpine` from dotnet variant
3. Version update to 0.20.3990

Fixes #5688

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [x] Bug fix
- [x] Package update

### CODE REVIEW

**Comment by mreid-tt** [comment]
hey @hgy59, no I haven't done any specific DSM 7.2 builds but this seems to be required for packages which include dotnet 6.x based on the error message (similarly reported issue with Sonarr v4). The user that reported the issue originally has since confirmed that with the addition of `libicu`, their issue was resolved.

I don't know if this means that DSM 7.2 does not contain `libicu` whereas DSM 7.1 did or if something with dotnet 6.x behaves differently under the beta version of DSM 7.2. There was an alternate solution proposed to set the env with `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1` however that didn't seem to work for Jackett (although it did work for Sonarr v4).

**Comment by mreid-tt** [comment]
@hgy59, I've gone back and formally tested this package under DSM 7.2-64216 for install, upgrade and functionality. Let me know if we are good to merge.

**Comment by mreid-tt** [comment]
@hgy59, I've implemented the modification for the dotnet dependencies you suggested. Let me know if this is in line with your thoughts. Seems to work fine under DSM 7.2 now without the explicit `libicu` inclusion.

File: spk/jackett/Makefile [hunk 1]
```diff
@@ -4,6 +4,7 @@ SPK_REV = 17
 SPK_ICON = src/jackett.png
 DSM_UI_DIR = app
 
+OPTIONAL_DEPENDS = cross/libstdc++
```
Comment (hgy59) [inline spk/jackett/Makefile:7]:
```suggestion
OPTIONAL_DEPENDS = cross/libstdc++ cross/curl cross/$(SPK_NAME)-mono
```

**Comment by hgy59** [comment]
@mreid-tt you forgot to publish packages for armada370, alpine and alpine4k for DSM 7.1.
To avoid three dedicated packages I created  #5720

Once  #5720 is merged, I will publish such a package for alpine and alpine4k for DSM 7.1.
_And I also like to publish packages for DSM 5 and SRM (that work with Mono)_

**Comment by mreid-tt** [comment]
@hgy59, thanks much for looking into this. I did notice some builds missing after the fact but couldn't understand why. All the jobs listed under build https://github.com/mreid-tt/spksrc/actions/runs/4804069827 seemed to be successfully generated and uploaded. Can you help me understand where I went wrong? Looking at #5720, other than moving the excluded archs to a variable, the logic seems to be the same.

**Comment by hgy59** [comment]
@mreid-tt the github build action does not build all packages.

e.g. it does not build for SRM archs (and my personal interest is to publish package for ppc853x-5.2 as long as it succeeds to build).
finally I try to build packages for DSM 5.2 - not only ppc853x, but also 88f6281 (ARMv5) and x86 as there are some models that have version 5.2 as latest available DSM.

The github build action is (originally) not ment to build the packages to deploy. It was introduced to have a fast response whether the cross build of different archs succeed.
So we removed the build of SRM archs as those are almost the same as ARMv7 but with different archs in the INFO file of the package (and another toolchain)

So I always create the packages to deploy on my build server locally.


In the very early days of spksrc there was no generic arch handling and every package was built for every arch available.
Then a generic arch for amd64 (x86_64) archs was created. I.e. a single toolchain was chosen to create a package for all x86_64 archs. To support several archs the only requirement in the package file is the list of archs in the INFO file.
One of my first contribution to SynoCommunity was to create a generic ARMv7 package. After analysing the different toolchains I created the armv7 arch based on the armada38x toolchain.

Bu now there are some issues with the generic ARMv7 archs, as some libraries (or frameworks like dotnet) are expecting some cpu specific features (like floating point capabilities)  that are not available on some arm archs.
So we started to exclude some arm32 archs from the generich ARMv7 builds.

Jacket is special in this case, as we have the option to use mono instead of dotnet.
As now armada370, alpine and alpine4k are removed from the dotnet version, built for ARMv7 generic arch, we have to create a package based on mono for all three archs.
To build only one package for the three (and to save disk space in the repo, and space is verly limited) we can build a package that is like a _generic_ package, built for multiple archs. Therefore we have the variable `SPK_PACKAGE_ARCHS`.
_(probably you only saw the variable `DSM7_ARMv7_ARCHS_DEPENDING_ON_MONO` that I introduced to define the list of archs only once)_

**Comment by hgy59** [comment]
BTW I have a DS115j with armada370 cpu.
I got the message, that a jackett update is available. But when updating, there is an error `invalid file format`.
Then I saw in the repo, that there is no package for armada370 and DSM 7.1.

A known limitation of the repo is, that it delivers DSM 6 packages for DSM 7 when no DSM 7 package is available. And this is the case here. The provided update is the package for armada370-6.1.

**Comment by mreid-tt** [comment]
> As now armada370, alpine and alpine4k are removed from the dotnet version, built for ARMv7 generic arch, we have to create a package based on mono for all three archs.
> To build only one package for the three (and to save disk space in the repo, and space is verly limited) we can build a package that is like a _generic_ package, built for multiple archs. Therefore we have the variable `SPK_PACKAGE_ARCHS`.

Okay so looking at the proposed code (formatting added) we have this:

```
DSM7_ARMv7_ARCHS_DEPENDING_ON_MONO = armada370 alpine alpine4k
```
```
ifeq ($(call version_lt, ${TCVERSION}, 7.0),1)
	ifeq ($(findstring $(ARCH),$(ARMv5_ARCHS) $(i686_ARCHS) $(ARMv7L_ARCHS) $(ARMv7_ARCHS)),$(ARCH))
		DEPENDS = cross/curl cross/jackett-mono
		SERVICE_SETUP = src/service-setup-mono.sh
		SPK_DEPENDS = "mono>=5.8"
	else
		DEPENDS = cross/jackett
		# dotnet requires libstdc++ version >= 6.0.22
		DEPENDS += cross/libstdc++
	endif
else
	ifeq ($(findstring $(ARCH),$(i686_ARCHS) $(DSM7_ARMv7_ARCHS_DEPENDING_ON_MONO)),$(ARCH))
		DEPENDS = cross/curl cross/jackett-mono
		SERVICE_SETUP = src/service-setup-mono.sh
		SPK_DEPENDS = "mono>=5.8"
		# create one package for all ARMv7 archs that depend on mono
		SPK_PACKAGE_ARCHS = $(DSM7_ARMv7_ARCHS_DEPENDING_ON_MONO)
	else
		DEPENDS = cross/jackett
		# exclude archs that depend on mono from generic package for ARMv7
		UNSUPPORTED_ARCHS += $(DSM7_ARMv7_ARCHS_DEPENDING_ON_MONO)
	endif
endif
```

Looking at the current [build run](https://github.com/SynoCommunity/spksrc/actions/runs/4852320573), we still only have one [Packages for armv7-6.1](https://github.com/SynoCommunity/spksrc/suites/12596672861/artifacts/673805549) and one [Packages for armv7-7.1](https://github.com/SynoCommunity/spksrc/suites/12596672861/artifacts/673805550). Digging into the INFO files we see the following:

_jackett_armv7-6.1_0.20.3990-17_
```
arch="alpine alpine4k armada370 armada375 armada38x armadaxp comcerto2k monaco"
os_min_ver="6.1-15047"
install_dep_packages="mono>=5.8"
```

_jackett_armv7-7.1_0.20.3990-17_
```
arch="armada375 armada38x armadaxp monaco"
os_min_ver="7.1-42661"
```

Does this mean we have to build a custom mono version for DSM 7.1? As you see above the existing `armv7-7.1` does not include `alpine alpine4k armada370`.

> I have a DS115j with armada370 cpu and DSM 7.1.1.
> I got the message, that a jackett update is available. But when updating, there is an error `invalid file format`.
> Then I saw in the repo, that there is no package for armada370 and DSM 7.1.

Based on the above, if you publish the current build run as-is my suspicion is that the same thing will happen on your `armada370`. It seems that we need to have two "Packages for armv7-7.1", one with dotnet and one with mono but the current Makefile does not seem to produce these.

**Comment by hgy59** [comment]
> Based on the above, if you publish the current build run as-is my suspicion is that the same thing will happen on your `armada370`. It seems that we need to have two "Packages for armv7-7.1", one with dotnet and one with mono but the current Makefile does not seem to produce these.

we do have to build a package for jackett depending on mono

_jackett_xxxxxxx-7.1_0.20.3990-17_

```
arch="armada370 alpine alpine4k"
os_min_ver="7.1-42661"
```

The file name can either be _jackett_armada370-7.1_0.20.3990-17_ or _jackett_alpine-7.1_0.20.3990-17_ or _jackett_alpine4k-7.1_0.20.3990-17_ it does not matter for the repository.

The name of the uploaded file is not used by the repository. As you can see when you manually download a file from the repo, it builds a file name containing all archs of the package.

This is solved with this PR. You can now build either for armada370-7.1 or alpine-7.1 or alpine4k-7.1 to create such a package.

**Comment by mreid-tt** [comment]
> we do have to build a package for jackett depending on mono
> 
> This is solved with this PR. You can now build either for armada370-7.1 or alpine-7.1 or alpine4k-7.1 to create such a package.

Okay but how is this to be accomplished? Do we still generate these builds using the GitHub build action? Would you change the inputs to this:

Screenshot 2023-05-01 at 3 35 05 PM

**Comment by hgy59** [comment]
@mreid-tt as I mentioned above, I never use this to publish packages.

**Comment by mreid-tt** [comment]
> @mreid-tt as I mentioned above, I never use this to publish packages.

Ahh, okay. I now understand that it must be built manually from a local instance per this comment...

> So I always create the packages to deploy on my build server locally.

If this is the case then I believe in the long term we will need to:
1. Add to the Wiki documentation to provide details on the recommended build process for publishing. 
Optionally add comments to this package that advises it should not be published using GitHub
2. Alternately, enhance the spksrc itself to provide additional options to handle complex build scenarios that can be executed within GitHub

For now, I won't attempt to publish this special package myself until I have a local setup and better understand the publishing intricacies. Do I need to be concerned about other packages I've built and published (possibly missing builds) or is Jackett the only special case?

**Comment by Stanzilla** [comment]
Are there two jackett packages now? When I tried to update today it prompted me to install mono again but I'm pretty sure I was on the .net core version?

**Comment by mreid-tt** [comment]
> Are there two jackett packages now? When I tried to update today it prompted me to install mono again but I'm pretty sure I was on the .net core version?

No, I don't think that is what @hgy59 was saying. With the current repository, using the GitHub build action, it is not currently capable of building for multiple `cross/*` builds within the same DSM version. What this means is that for example with DSM 7.x the default build will be either the dotnet build or the mono build but not both. To build both (in future publishing actions), the builds will need to be created manually so that each can be uploaded for publishing.

As for what you are experiencing, you mentioned in this https://github.com/SynoCommunity/spksrc/issues/2808#issuecomment-344403216, that you have a DS212 which is an ARMv5 arch. If this is the case, it would require mono for installation. If you are on a platform that does support dotnet and it is installing the wrong version I'd recommend opening a new issue with the details on the platform you are running on.

**Comment by Stanzilla** [comment]
I see, because I was looking at https://synocommunity.com/package/jackett and tried to find a 7.1 x64 package.

**Pull request merged**